tools/src/main/java/org/apache/kafka/tools/SchemaChecker/MetadataSchemaChecker.java

```diff
@@ -0,0 +1,347 @@
+        Repository repository = git.getRepository();
+        Ref head = git.getRepository().getRefDatabase().firstExactRef("refs/heads/trunk");
+
+        try (RevWalk revWalk = new RevWalk(repository)) {
+            RevCommit commit = revWalk.parseCommit(head.getObjectId());
+            RevTree tree = commit.getTree();
+            for (String jsonSchema : filesCheckMetadata) {
+                StringBuilder stringBuilder = new StringBuilder();
+                try (TreeWalk treeWalk = new TreeWalk(repository)) {
+                    treeWalk.addTree(tree);
+                    treeWalk.setRecursive(true);
+                    treeWalk.setFilter(PathFilter.create("metadata/src/main/resources/common/metadata/" + jsonSchema));
+                    if (!treeWalk.next()) {
+                        throw new IllegalStateException("Did not find expected file /metadata/src/main/resources/common/metadata/" + jsonSchema);
+                    }
+                    ObjectId objectId = treeWalk.getObjectId(0);
+                    ObjectLoader loader = repository.open(objectId);
+
+                    String content = new String(loader.getBytes());
+                    String[] lines = content.split("\n");
+                    boolean print = false;
+                    for (int i = 15; i < lines.length; i++) {
+                        if (lines[i].charAt(0) == '{') {
+                            print = true;
+                        }
+                        if (print && !lines[i].contains("//")) {
+                            stringBuilder.append(lines[i]);
+                        }
+                    }
+                }
+                gitSchemas.add(stringBuilder.toString());
+            }
+            revWalk.dispose();
+        }
+        return gitSchemas;
+    }
+
+    private static void checkApiTypeVersions(JsonNode original, JsonNode edited) {
+        if (!Objects.equals(original.get("apiKey"), edited.get("apiKey"))) {
+            throw new IllegalStateException("New schema has wrong api key, " + edited.get("apiKey") + " should be " + original.get("apiKey"));
+        }
+        if (!Objects.equals(original.get("type"), edited.get("type"))) {
+            throw new IllegalStateException("New schema has wrong record type, " + edited.get("type") + " should be " + original.get("type"));
+        }
+        String oldValidVersions = String.valueOf(original.get("validVersions"));
+        String newValidVersions = String.valueOf(edited.get("validVersions"));
+        oldLatestVersion = Character.getNumericValue(oldValidVersions.charAt(oldValidVersions.length() - 2));
+        newLatestVersion = Character.getNumericValue(newValidVersions.charAt(newValidVersions.length() - 2));
+        if (oldLatestVersion != newLatestVersion && oldLatestVersion + 1 != newLatestVersion) {
+            throw new IllegalStateException("Invalid latest versions, can at most be one higher than the previous");
```

Comment:
Invalid latest version (singular not plural)

tell us what the versions were that had the problem